snippets/epel.erb

```diff
@@ -1,6 +1,10 @@
 <%#
 kind: snippet
 name: epel
-%>
-
-su -c 'rpm -Uvh <%= @host.os.medium_uri(@host, "http://dl.fedoraproject.org/pub/epel/epel-release-latest-#{@host.operatingsystem.major}.noarch.rpm") %>'
+-%>
+<%
+  http_proxy = @host.params['http-proxy'] ? " --httpproxy #{@host.params['http-proxy']}" : nil
+  http_port = @host.params['http-proxy-port'] ? " --httpport #{@host.params['http-proxy-port']}" : nil
+  os_major = @host.operatingsystem.major.to_i
+-%>
+rpm -Uvh<%= http_proxy %><%= http_port %> <%= @host.os.medium_uri(@host, "http://dl.fedoraproject.org/pub/epel/epel-release-latest-#{os_major}.noarch.rpm") %>
```

Comment:
It might be worth changing the URLs to HTTPS while you're here, they all support it now and it's important when downloading a release key.